src/Microsoft.Framework.Runtime/DefaultHost.cs

```diff
@@ -58,11 +60,34 @@ public Assembly GetEntryPoint(string applicationName)
 
             Initialize();
 
+            var unresolvedLibs = _applicationHostContext.DependencyWalker.Libraries.Where(l => !l.Resolved);
+
             // If there's any unresolved dependencies then complain
-            if (_applicationHostContext.DependencyWalker.Libraries.Any(l => !l.Resolved))
+            if (unresolvedLibs.Any())
             {
-                var exceptionMsg = _applicationHostContext.DependencyWalker.GetMissingDependenciesWarning(
-                    _targetFramework);
+                string exceptionMsg;
+
+                // If the main project cannot be resolved, it means the app doesn't support current target framework
+                // (i.e. project.json doesn't contain a framework that is compatible to target framework of current runtime)
+                if (unresolvedLibs.Any(l => string.Equals(l.Identity.Name, Project.Name)))
+                {
+                    var runtimeEnv = ServiceProvider.GetService(typeof(IRuntimeEnvironment)) as IRuntimeEnvironment;
+                    var alternativeType = VersionUtility.IsDesktop(_targetFramework) ? "CoreCLR" : "CLR";
+                    var shortName = VersionUtility.GetShortFrameworkName(_targetFramework);
+                    exceptionMsg = string.Join(Environment.NewLine, new[]{
+                        $"'{Project.Name}' doesn't support current runtime target framework." + Environment.NewLine,
+                        $"Runtime Target Framework: '{_targetFramework} ({shortName})'",
+                        $"Runtime Type: {runtimeEnv.RuntimeType}",
+                        $"Runtime Architecture: {runtimeEnv.RuntimeArchitecture}",
+                        $"Runtime Version: {runtimeEnv.RuntimeVersion}" + Environment.NewLine,
+                        $"Please make sure the runtime matches a framework specified in {Project.ProjectFileName}" });
```

Comment:
I discussed this with @davidfowl yesterday. There are two possible solutions when a user runs into this:
1. Switch to another flavor of runtime
2. Add proper framework into `frameworks` section of `project.json`

We don't which one the best suggestion because both of them are possible. So we came up with a summary that covers both case:
`Please make sure the runtime matches a framework specified in project.json`